### PR TITLE
Multiple code style improvements

### PR DESCRIPTION
**Pull request by paskal**
- IDE lint fixes
- typos fixes
- add golangci-lint config, fix discovered issues
- add GutHub Actions workflow for tests run

Please let me know if you want to exclude some changes from this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,12 +134,12 @@ func main() {
     lmt.SetOnLimitReached(func(w http.ResponseWriter, r *http.Request) { fmt.Println("A request was rejected") })
     ```
 
-6. Tollbooth does not require external storage since it uses an algorithm called [Token Bucket](http://en.wikipedia.org/wiki/Token_bucket) [(Go library: golang.org/x/time/rate)](//godoc.org/golang.org/x/time/rate).
+6. Tollbooth does not require external storage since it uses an algorithm called [Token Bucket](http://en.wikipedia.org/wiki/Token_bucket) [(Go library: golang.org/x/time/rate)](https://godoc.org/golang.org/x/time/rate).
 
 
 ## Other Web Frameworks
 
-Sometimes, other frameworks require a little bit of shim to use Tollbooth. These shims below are contributed by the community, so I make no promises on how well they work. The one I am familiar with are: Chi, Gin, and Negroni.
```
Comment (didip) [inline README.md:143]: `a little of shim` sounds a tad strange, no?

**Comment by paskal** [comment]
I found `Test_Issue48_RequestTerminatedEvenOnLowVolumeOnSameIP` failing in master and two more failing only in this branch, working on fix.

**Comment by paskal** [comment]
Now it's working.

Issue48 test is failing in test and failing [in the pipeline](https://erudit.netlify.app/room/151) of this branch as well.

File: README.md (hunk 1 above)
Comment (paskal) [reply README.md]: This was reverted, thanks for noticing. Would you be so kind as to look into it once more?

**Comment by paskal** [comment]
@didip would you be so kind as to look into it once more?

File: tollbooth.go [hunk 2]
```diff
@@ -159,7 +158,7 @@ func LimitHandler(lmt *limiter.Limiter, next http.Handler) http.Handler {
 			lmt.ExecOnLimitReached(w, r)
 			w.Header().Add("Content-Type", lmt.GetMessageContentType())
 			w.WriteHeader(httpError.StatusCode)
-			w.Write([]byte(httpError.Message))
+			_, _ = w.Write([]byte(httpError.Message))
```
Comment (didip) [inline tollbooth.go:161]: Is this necessary?

File: tollbooth_bug_report_test.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func Test_Issue48_RequestTerminatedEvenOnLowVolumeOnSameIP(t *testing.T) {
 	lmt.SetOnLimitReached(func(w http.ResponseWriter, r *http.Request) { limitReachedCounter++ })
 
 	handler := LimitHandler(lmt, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`hello world`))
+		_, _ = w.Write([]byte(`hello world`))
```
Comment (didip) [inline tollbooth_bug_report_test.go:21]: is this necessary?

File: tollbooth_test.go [hunk 4]
```diff
@@ -345,7 +347,7 @@ func TestLimitHandler(t *testing.T) {
 	lmt.SetOnLimitReached(func(w http.ResponseWriter, r *http.Request) { counter++ })
 
 	handler := LimitHandler(lmt, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`hello world`))
+		_, _ = w.Write([]byte(`hello world`))
```
Comment (didip) [inline tollbooth_test.go:350]: Is this coming from automated linter?

File: tollbooth.go (hunk 2 above)
Comment (paskal) [reply tollbooth.go:161]: It's an explicit way to tell whoever reads the code "this function returns an error but I don't care about it and discard it".

Comment (paskal) [reply tollbooth.go:161]: I can revert it if you want, it's just a common convention. In https://github.com/golang/go.git for example:
```
➜  go git:(master) fgrep '_, _ =' . -r | grep -v "_test.go" | wc -l
     168
➜  go git:(master) fgrep '_ =' . -r | grep -v "_test.go" | wc -l
    7315
```

Comment (didip) [reply tollbooth.go:161]: Let's do that :)

Comment (paskal) [reply tollbooth.go]: Done, I'll resolve these threads.

**Pull request merged**